sdk/nodejs/cmd/run-plugin/index.ts

```diff
@@ -0,0 +1,127 @@
+        console.error(err.stack || err.message || "" + err);
+    }
+};
+
+// Keep track if we already logged the information about an unhandled error to the user..  If
+// so, we end with a different exit code.  The language host recognizes this and will not print
+// any further messages to the user since we already took care of it.
+//
+// 32 was picked so as to be very unlikely to collide with any of the error codes documented by
+// nodejs here:
+// https://github.com/nodejs/node-v0.x-archive/blob/master/doc/api/process.markdown#exit-codes
+const nodeJSProcessExitedAfterLoggingUserActionableMessage = 32;
+
+process.on("uncaughtException", uncaughtHandler);
+// @ts-ignore 'unhandledRejection' will almost always invoke uncaughtHandler with an Error. so just
+// suppress the TS strictness here.
+process.on("unhandledRejection", uncaughtHandler);
+process.on("exit", (code: number) => {
+    // If there were any uncaught errors at all, we always want to exit with an error code. If we
+    // did not, it could be disastrous for the user.  i.e. not all resources may have been created,
+    // but the 0 code would indicate we could proceed.  That could lead to many (or all) of the
+    // user resources being deleted.
+    if (code === 0 && uncaughtErrors.size > 0) {
+        // Now Check if this error was already logged to the user in a visible fashion.  If not
+        // we will exit with '1', indicating that the host should give a generic message about
+        // things not working.
+        for (const err of uncaughtErrors) {
+            if (!loggedErrors.has(err)) {
+                process.exitCode = 1;
+                return;
+            }
+        }
+
+        process.exitCode = nodeJSProcessExitedAfterLoggingUserActionableMessage;
+    }
+});
+
+// As the second thing we do, ensure that we're connected to v8's inspector API.  We need to do
+// this as some information is only sent out as events, without any way to query for it after the
+// fact.  For example, we want to keep track of ScriptId->FileNames so that we can appropriately
+// report errors for Functions we cannot serialize.  This can only be done (up to Node11 at least)
+// by register to hear about scripts being parsed.
+import * as v8Hooks from "../../runtime/closure/v8Hooks";
+
+// This is the entrypoint for running a Node.js program with minimal scaffolding.
+import minimist from "minimist";
+
+function usage(): void {
+    console.error(`usage: RUN <program>`);
+}
+
+function printErrorUsageAndExit(message: string): never {
+    console.error(message);
+    usage();
+    return process.exit(-1);
+}
+
+function main(args: string[]): void {
+    // See usage above for the intended usage of this program, including flags and required args.
+    const argv: minimist.ParsedArgs = minimist(args, {});
+
+    // Finally, ensure we have a program to run.
+    if (argv._.length < 1) {
+        return printErrorUsageAndExit("error: Usage: RUN <program>");
```

Comment:
Looks like this will double print the usage message, no? Maybe worth cleaning this code up.

Reply:
Yes, you're right.  Will fix!